src/test/java/uk/gov/hmcts/ccd/domain/service/casefileview/CategoriesAndDocumentsServiceTest.java

```diff
@@ -254,19 +254,36 @@ void testShouldRaiseExceptionWhenInputIsNotNullButInvalidTimestamp() {
             .isInstanceOf(RuntimeException.class);
     }
 
-    @Test
-    void testShouldParseTimestamp() {
-        final String timestamp = "2022-04-06T16:44:52.000000000";
-        final LocalDateTime expectedTimestamp = LocalDateTime.of(2022, 4, 6, 16,
-            44, 52);
-
+    @ParameterizedTest
+    @MethodSource("provideTimestampParameters")
+    void testShouldParseTimestamps(String timestamp, LocalDateTime expectedTimestamp) {
         final LocalDateTime result = underTest.parseUploadTimestamp(timestamp);
 
         assertThat(result)
             .isNotNull()
             .isEqualTo(expectedTimestamp);
     }
 
+    private static Stream<Arguments> provideTimestampParameters() {
+        return Stream.of(
+            Arguments.of(
+                "2022-04-06T16:44:52.000000000Z", LocalDateTime.of(2022, 4, 6, 16, 44, 52)
+            ),
+            Arguments.of(
+                "2022-04-06T16:44:52.000000000", LocalDateTime.of(2022, 4, 6, 16, 44, 52)
```

Comment:
Prior to the changes in this PR, this example is the only one of the now 5 tests which passed.